Doc/reference/datamodel.rst

```diff
@@ -1512,6 +1512,46 @@ access (use of, assignment to, or deletion of ``x.name``) for class instances.
    returned. :func:`dir` converts the returned sequence to a list and sorts it.
 
 
+Customizing module attribute access
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Special names ``__getattr__`` and ``__dir__`` can be also used to customize
+access to module attributes. The ``__getattr__`` function at the module level
+should accept one argument which is the name of an attribute and return the
+computed value or raise an :exc:`AttributeError`. If an attribute is
+not found on a module object through the normal lookup, i.e.
+:meth:`object.__getattribute__`, then ``__getattr__`` is searched in
+the module ``__dict__`` before raising an :exc:`AttributeError`. If found,
```

Comment:
Would be nice to make a reference to `__dict__` as the module attribute (but not to the object attribute) if possible.

Reply:
I am not sure if it is possible. The sub-section "Modules" itself refers to ``object.__dict__``.

Reply:
There is an index entry for `__dict__` as the module attribute. But I don't know if it is possible to refer to the target of an index entry.

Reply:
I believe you can you use ``:data:`` directive.  Here's the definition in the Documentation Guide:
```
data

    Describes global data in a module, including both variables and values used as
   “defined constants.” Class and object attributes are not documented using this directive.
```

Where the ``__dict__`` is defined:
```
   .. index:: single: __dict__ (module attribute)
   .. data:: __dict__

   Special read-only attribute: :attr:`~object.__dict__` is the module's
   namespace as a dictionary object.
```

and then where you want to link:
```
:meth:`object.__getattribute__`, then ``__getattr__`` is searched in
the module :data:`__dict__` before raising an :exc:`AttributeError`.
```

As you mentioned, this may require fixing other links for ``module.__dict__`` that now point to ``object.__dict__``.

Reply:
I'm not sure this is applicable here. In any case we can experiment with this in a separate issue. I prefer to keep the current markup for now.

Reply:
> In any case we can experiment with this in a separate issue. I prefer to keep the current markup for now.

I also think it can be deferred to a separate issue. IIUC the only thing remaining here is to get an OK from @ncoghlan